src/protocol/requests/listGroups/v3/response.js

```diff
@@ -0,0 +1,12 @@
+const responseV2 = require('../v2/response')
+
+/**
+ * ListGroups Response (Version: 3) => throttle_time_ms error_code [groups] TAG_BUFFER
+ *   throttle_time_ms => INT32
+ *   error_code => INT16
+ *   groups => group_id protocol_type TAG_BUFFER
+ *     group_id => COMPACT_STRING
+ *     protocol_type => COMPACT_STRING
+ */
+
+module.exports = responseV2
```

Comment:
I don't think this will work, because v2 and below give `group_id` and `protocol_type` as strings, but these are compact strings that need to be read differently. I'm not even sure if we have implemented support for those yet - unless it's what we refer to as a `VarIntString` in src/protocol/decoder.js.

Reply:
Should I remove v3 then?

Reply:
I think `readVarIntString` in the decoder will actually work, if I'm reading the spec right. Try implementing `decode` in v3 to use `readVarIntString` instead of `readString` for `groupId` and `protocolType`.